doc/api/process.md

```diff
@@ -3303,6 +3303,30 @@ In custom builds from non-release versions of the source tree, only the
 `name` property may be present. The additional properties should not be
 relied upon to exist.
 
+## \`process.replace(file\[, args\[, env]])\`
+
+<!-- YAML
+added: REPLACEME
+-->
+
+* `file` {string} The name or path of the executable file to run.
+* `args` {string\[]} List of string arguments. No argument can contain a null-byte (`\u0000`).
+* `env` {Object} Environment key-value pairs.
+  No key or value can contain a null-byte (`\u0000`).
+  **Default:** `process.env`.
+
+Replaces the current process with a new process.
+
+This is achieved by using the `execve` Unix function and therefore no memory or other
+resources from the current process are preserved, except for the standard input,
+standard output and standard error file descriptor.
+
+All other resources are discarded by system when the processes are swapped.
```

Comment:
What if this isn't the a desirable behavior in a given situation?

Reply:
Can you please expand this? What do you mean?

Reply:
If there's any point in adding execve(), it's the customizability that the method brings with it. Leaving file descriptors open  and/or redirecting them intentionally can be part of that -- just like you could in a bash script do `exec bash 0<&4 1>&4` to create a shell that reads and writes to e.g. a pre-opened socket or something along those lines.

(With a similar reasoning, you could also allow users to actually specify `argv[0]` with a value they control -- it doesn't _need_ to equal the filename that's being executed)